socs/agents/srs_cg635/agent.py

```diff
@@ -0,0 +1,237 @@
+import argparse
+import socket
+import time
+from os import environ
+
+import txaio
+from ocs import ocs_agent, site_config
+from ocs.ocs_twisted import TimeoutLock
+
+from socs.agents.srs_cg635.drivers import SRSCG635Interface
+
+
+class SRSCG635Agent:
+    """Class to retrieve data from the SRS CG635 clock.
+
+    Parameters
+    ----------
+    agent : ocs_agent.OCSAgent
+        Instantiated OCSAgent class for this Agent.
+    ip_address : str
+        IP address of the Prologix GPIB interface.
+    gpib_slot : int
+        GPIB address set on the SRS CG635.
+    """
+
+    def __init__(self, agent, ip_address, gpib_slot):
+        self.agent = agent
+        self.log = agent.log
+        self.lock = TimeoutLock()
+
+        self.job = None
+        self.ip_address = ip_address
+        self.gpib_slot = gpib_slot
+        self.monitor = False
+
+        self.clock = None
+
+        # Registers Temperature and Voltage feeds
+        agg_params = {
+            'frame_length': 10 * 60,
+        }
+        self.agent.register_feed('srs_clock',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=0)
+
+    @ocs_agent.param('auto_acquire', default=False, type=bool)
+    def init(self, session, params=None):
+        """init(auto_acquire=False)
+
+        **Task** - Initialize the connection to the srs clock.
+
+        Parameters
+        ----------
+        auto_acquire: bool, optional
+            Default is False. Starts data acquisition after initialization
+            if True.
+        """
+        with self.lock.acquire_timeout(0) as acquired:
+            if not acquired:
+                return False, "Could not acquire lock"
+
+            try:
+                self.clock = SRSCG635Interface(self.ip_address, self.gpib_slot)
+                self.idn = self.clock.identify()
+
+            except (socket.timeout, OSError) as e:
+                self.log.error(f"Clock timed out during connect: {e}")
+                return False, "Timeout"
+            self.log.info("Connected to Clock: {}".format(self.idn))
+
+        # Start data acquisition if requested in site-config
+        auto_acquire = params.get('auto_acquire', False)
```

Comment:
Since this is decorated you can just call `params['auto_acquire']`. Can probably just delete this line and call `if params['auto_acquire']:`